src/Http/Http.Results/test/ResultsTests.cs

```diff
@@ -1268,6 +1268,30 @@ public void Problem_WithArgs_ResultHasCorrectValues()
         Assert.Equal(extensions, result.ProblemDetails.Extensions);
     }
 
+    [Fact]
+    public void Problem_ResultHasCorrectValues()
+    {
+        // Arrange
+        var detail = "test detail";
+        var instance = "test instance";
+        var statusCode = StatusCodes.Status409Conflict;
+        var title = "test title";
+        var type = "test type";
+        var extensions = new List<KeyValuePair<string, object>> { new("test", "value") };
```

Comment:
```suggestion
        var extensions = (IReadOnlyDictionary<string, object>)new Dictionary<string, object> { ["test"] = "value" };
```

It'd be nice to verify it with an `IReadOnlyDictionary` since that's the primary use case for these new overloads. I also understand wanting to test with non-dictionaries to catch any potential issues where we might try to unsafely cast to a dictionary. We can just add a `Problem_WithReadOnlyDictionary_ResultHasCorrectValues` variation.

Reply:
Done.